src/scripts/main.js

```diff
@@ -1,3 +1,337 @@
+  const positionValue = form.elements['position'].value;
+  const selectedOffice = form.elements['city'].value;
+  const ageValue = Number(form.elements['age'].value);
+  const salaryValue = Number(form.elements['salary'].value);
+
+  // Form validation checks
+  if (!form.checkValidity()) {
+    showNotification('error', 'Please fill in all required fields.');
+
+    return;
+  }
+
+  if (nameValue.length < MIN_NAME_LENGTH) {
+    showNotification('error', 'Name must be at least 4 characters long.');
+
+    return;
+  }
+
+  if (ageValue < MIN_AGE || ageValue > MAX_AGE) {
+    showNotification('error', 'Age must be between 18 and 90.');
+
+    return;
+  }
+
+  // Create new employee row and add to table
+  const newEmployee = createNewEmployee(
+    nameValue,
+    positionValue,
+    selectedOffice,
+    ageValue,
+    salaryValue,
+  );
+
+  // Notify user of successful form submission
+  showNotification('success', 'Form submitted successfully!');
+  tableBody.append(newEmployee);
+
+  // Reset the form fields
+  form.reset();
+}
+
+// Creates a new employee row with the provided data
+function createNewEmployee(employeName, position, office, age, salary) {
+  const tableRow = document.createElement('tr');
+
+  // Format salary as currency
+  const formattedSalary = salary.toLocaleString('en-US', {
+    style: 'currency',
+    currency: 'USD',
+    minimumFractionDigits: 0,
+    maximumFractionDigits: 0,
+  });
+
+  // Create an array of table data and loop through it
+  const tableData = [employeName, position, office, age, formattedSalary];
+
+  tableData.forEach((dataText) => {
+    const td = document.createElement('td');
+
+    td.innerText = dataText; // Insert each piece of data into the cell
+    tableRow.append(td);
+  });
+
+  return tableRow;
+}
+
+// Creates a label element with the provided text and data attribute
+function createLabel(text, data) {
+  const label = document.createElement('label');
+
+  label.setAttribute('data-qa', data);
+  label.innerHTML = `${text}&nbsp`; // Add a non-breaking space after the text
+
+  return label;
+}
+
+// Creates an input element with the provided name and type
+function createInput(inputName, InputType) {
+  const input = document.createElement('input');
+
+  input.setAttribute('name', inputName);
+  input.setAttribute('type', InputType);
+  input.setAttribute('required', ''); // Make the input field required
+
+  return input;
+}
+
+// Creates a select element with options
+function createSelect(options) {
+  const select = document.createElement('select');
+
+  select.setAttribute('required', ''); // Make the select field required
+
+  // Add city options to the select dropdown
+  options.forEach((city) => {
+    const option = document.createElement('option');
+
+    option.textContent = city;
+    option.value = city;
+    select.append(option);
+  });
+
+  return select;
+}
+
+// Creates a submit button for the form
+function createSubButton(formId, text) {
```

Comment:
Consider renaming the function `createSubButton` to `createSubmitButton` for better clarity and to accurately reflect its purpose.